### PR TITLE
Add sample for firmware update

### PR DESCRIPTION
**Pull request by torbsorb**
This sample shows how to use the APIs to check whether a camera requires an updated, and update if required.

### CODE REVIEW

File: source/Camera/InfoUtilOther/FirmwareUpdater/FirmwareUpdater.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        auto cameras = zivid.cameras();
+
+        if(cameras.empty())
+        {
+            std::cout << "No camera(s) found." << std::endl;
```
Comment (nedrebo) [inline source/Camera/InfoUtilOther/FirmwareUpdater/FirmwareUpdater.cpp]: `return EXIT_FAILURE;` here hand drop the else.

**Comment by torbsorb** [comment]
This sample is a copy of https://github.com/zivid/zivid-sdk/blob/master/cpp/CLI/FirmwareUpdater/FirmwareUpdater.cpp. A sample output is:

```bash
$ ./ZividFirmwareUpdater.exe
Found 1 camera(s).
Skipping update of camera 192445D3, model name: Zivid One Plus M, firmware version: 1.6.7
```

**Comment by nedrebo** [comment]
I know its a copy, but when elevating something to public release it is worth additional love. All internal code is not ready to be shown publicly.

What is the output when actually flashing FW?

**Comment by torbsorb** [comment]
@nedrebo 
```
47481@DESKTOP-PM9L3T4 MINGW64 /c/Program Files/Zivid_1_6_0/bin
$ ./ZividFirmwareUpdater.exe
Found 1 camera(s).
Updating firmware on camera 192445D3, model name: Zivid One Plus M, firmware version: 1.6.7
0 % : Checking camera state...
5 % : Checking existing version...
10 % : Retrieving internal firmware...
15 % : Updating firmware on camera...
20 % : Getting device to flash...
24 % : Checking that firmware is verified...
28 % : Writing firmware image...
36 % : Rebooting device...
88 % : Reconnecting to device...
92 % : Checking firmware version...
96 % : Setting firmware as verified...
100 % : Firmware successfully updated to version 1.6.6

47481@DESKTOP-PM9L3T4 MINGW64 /c/Program Files/Zivid_1_6_0/bin
$ cd ../../Zivid/bin

47481@DESKTOP-PM9L3T4 MINGW64 /c/Program Files/Zivid/bin
$ ./ZividFirmwareUpdater.exe
Found 1 camera(s).
Updating firmware on camera 192445D3, model name: Zivid One Plus M, firmware version: 1.6.6
0 % : Checking camera state...
5 % : Checking existing version...
10 % : Retrieving internal firmware...
15 % : Updating firmware on camera...
20 % : Getting device to flash...
24 % : Checking that firmware is verified...
28 % : Writing firmware image...
36 % : Rebooting device...
88 % : Reconnecting to device...
92 % : Checking firmware version...
96 % : Setting firmware as verified...
100 % : Firmware successfully updated to version 1.6.7

47481@DESKTOP-PM9L3T4 MINGW64 /c/Program Files/Zivid/bin
$
```

**Pull request merged**